### PR TITLE
Improve error handling for diesel_cli

### PR DESCRIPTION
**Pull request by mcasper**
Improve error handling for diesel cli by implementing `std::error::Error` for migration related error structs.

### CODE REVIEW

File: diesel/src/migrations/migration_error.rs [hunk 1]
```diff
@@ -12,6 +15,23 @@ pub enum MigrationError {
     UnknownMigrationVersion(String),
 }
 
+impl Error for MigrationError {
+    fn description(&self) -> &str {
+        match *self {
+            MigrationDirectoryNotFound => "Unable to find migrations directory in this directory or any parent directories.",
+            UnknownMigrationFormat(_) => "Invalid migration directory, the directory's name should be <timestamp>_<name_of_migration>, and it should only contain up.sql and down.sql.",
+            IoError(ref error) => error.description(),
+            UnknownMigrationVersion(_) => "Invalid migration version, only YYYYMMDDHHMMSS format is supported.",
```
Comment (sgrif) [inline diesel/src/migrations/migration_error.rs]: That's not what this means. It occurs during revert if we can't find the migration on disk that the database says we should revert

File: diesel/src/migrations/migration_error.rs [hunk 2]
```diff
@@ -12,6 +15,23 @@ pub enum MigrationError {
...
+        }
+    }
+}
+
+impl fmt::Display for MigrationError {
```
Comment (sgrif) [inline diesel/src/migrations/migration_error.rs]: Does error require display?

Comment (mcasper) [inline diesel/src/migrations/migration_error.rs]: Yep
https://doc.rust-lang.org/std/error/

File: diesel/src/migrations/migration_error.rs (hunk 1 above)
Comment (mcasper) [inline diesel/src/migrations/migration_error.rs]: Ahhh okay, I'll change the description

**Comment by sgrif** [comment]
Can you squash your commits and rebase so the build goes green?

**Comment by mcasper** [comment]
Rebased and green!

**Comment by sgrif** [comment]
Thanks. As a note, I'd appreciate someone taking the changes here further and removing the `.unwrap` calls since we have a decent way to handle these at the top level. I'm not huge on the `exit(1)` in `map_err` calls, but it's better than what we have now

**Pull request merged**